File: exercises/py101/easy3/bannerize.py


def print_in_box(text):
    horizontal_rule = f'+-{'-' * len(text)}-+'
    empty_rule = f'| {' ' * len(text)} |'

    print(horizontal_rule)
    print(empty_rule)
    print(f'| {text} |')
    print(empty_rule)
    print(horizontal_rule)

print_in_box('To boldly go where no one has gone before.')
print_in_box('')

File: exercises/js210/easy2/bannerize.js
function repeatString(str, times) {
  return str.repeat(times + 2);
}

function logInBox(sentence) {
  let strLength = sentence.length;
  let result = '';
  result += `+${repeatString('-', strLength)}+\n`;
  result += `|${repeatString(' ', strLength)}|\n`;
  result += `| ${sentence} |\n`;
  result += `|${repeatString(' ', strLength)}|\n`;
  result += `+${repeatString('-', strLength)}+`;

  return result;
}

console.log(logInBox('To boldly go where no one has gone before.'));
console.log(logInBox(''));